static/app/components/modals/privateGamingSdkAccessModal.tsx

```diff
@@ -0,0 +1,184 @@
+import {type ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/core/button';
+import {ButtonBar} from 'sentry/components/core/button/buttonBar';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import TextField from 'sentry/components/forms/fields/textField';
+import {t, tct} from 'sentry/locale';
+import type {Organization} from 'sentry/types/organization';
+import {useUser} from 'sentry/utils/useUser';
+
+const PRIVATE_GAMING_SDK_OPTIONS = [
+  {value: 'playstation', label: 'PlayStation'},
+  {value: 'xbox', label: 'Xbox'},
+  {value: 'nintendo-switch', label: 'Nintendo Switch'},
+] as const;
+
+type GamingPlatform = (typeof PRIVATE_GAMING_SDK_OPTIONS)[number]['value'];
+
+export interface PrivateGamingSdkAccessModalProps {
+  organization: Organization;
+  projectSlug: string;
+  sdkName: string;
+  gamingPlatform?: GamingPlatform;
+  onSubmit?: () => void;
+}
+
+export function PrivateGamingSdkAccessModal({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+  organization,
+  projectSlug,
+  sdkName,
+  gamingPlatform,
+  onSubmit,
+}: PrivateGamingSdkAccessModalProps & ModalRenderProps) {
+  const user = useUser();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [githubProfile, setGithubProfile] = useState('');
+  const [gamingPlatforms, setGamingPlatforms] = useState<string[]>(
+    gamingPlatform ? [gamingPlatform] : []
+  );
+
+  const isFormValid = !!githubProfile.trim() && gamingPlatforms.length > 0;
+
+  const handleSubmit = useCallback(() => {
+    if (!isFormValid) {
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    onSubmit?.();
+
+    const messageBody = [
+      `User: ${user.name}`,
+      `Email: ${user.email}`,
+      gamingPlatforms.length === 1
+        ? `Platform: ${gamingPlatforms[0]}`
+        : `Platforms: ${gamingPlatforms
+            .map(
+              (platform: string) =>
+                PRIVATE_GAMING_SDK_OPTIONS.find(option => option.value === platform)
+                  ?.label || platform
+            )
+            .join(', ')}`,
+      `Org Slug: ${organization.slug}`,
+      `Project: ${projectSlug}`,
+      `GitHub Profile: ${githubProfile}`,
+    ].join('\n');
+
+    const source = `${sdkName.toLowerCase()}-sdk-access`;
+
+    // Use captureFeedback with proper user context instead of tags
+    captureFeedback(
+      {
+        message: messageBody,
+        name: user.name,
+        email: user.email,
+        source,
+        tags: {
+          feature: source,
+        },
+      },
+      {
+        captureContext: {
+          user: {
+            id: user.id,
+            email: user.email,
+            username: user.username,
+            name: user.name,
+          },
+        },
+      }
+    );
+
+    addSuccessMessage(
+      tct('Your [sdkName] SDK access request has been submitted.', {
+        sdkName,
+      })
+    );
+    setIsSubmitting(false);
```

Comment:
I’m unsure how this should work. This `handleSubmit` function is a synchronous callback, so it first calls `setIsSubmitting(true)` followed by `setIsSubmitting(false)`, so I don’t see where the `true` state would ever be rendered 🤔 .

This is part of the reason why merging something that has no usage is a bit counter productive imo because we can’t know how it works / what it does.

I’m guessing `onSubmit` is going to maybe trigger a mutation on the call-side? Would it be better to trigger that mutation from inside here, which would also give us mutation state like `isPending` and callbacks to add the success messages properly?

Reply:
I agree - it would be great to use `isPending` and show a success message only if everything went well. But we're using `captureFeedback`, which doesn't wait for the result. It just tries to send the feedback and returns immediately. You can read more about it [here](https://github.com/getsentry/sentry-javascript/pull/11428).

This modal will be removed in a future phase, but for now, we’re keeping it simple by reusing what we already have.